include/PetriEngine/PQL/Visitor.h

```diff
@@ -368,116 +329,17 @@ namespace PetriEngine {
                 // no-op
             }
 
-            void _accept(const EFCondition *condition) override
-            {
-                (*condition)[0]->visit(*this);
-            }
-
-            void _accept(const EGCondition *condition) override
-            {
-                (*condition)[0]->visit(*this);
-            }
-
-            void _accept(const AGCondition *condition) override
-            {
-                (*condition)[0]->visit(*this);
-            }
-
-            void _accept(const AFCondition *condition) override
-            {
-                (*condition)[0]->visit(*this);
-            }
-
-            void _accept(const EXCondition *condition) override
-            {
-                (*condition)[0]->visit(*this);
-            }
-
-            void _accept(const AXCondition *condition) override
-            {
-                (*condition)[0]->visit(*this);
-            }
-
-            void _accept(const EUCondition *condition) override
-            {
-                (*condition)[0]->visit(*this);
-                (*condition)[1]->visit(*this);
-            }
-
-            void _accept(const AUCondition *condition) override
-            {
-                (*condition)[0]->visit(*this);
-                (*condition)[1]->visit(*this);
-            }
-
-            void _accept(const ACondition *condition) override
-            {
-                (*condition)[0]->visit(*this);
-            }
-
-            void _accept(const ECondition *condition) override
-            {
-                (*condition)[0]->visit(*this);
-            }
-
-            void _accept(const GCondition *condition) override
-            {
-                (*condition)[0]->visit(*this);
-            }
-
-            void _accept(const FCondition *condition) override
-            {
-                (*condition)[0]->visit(*this);
-            }
-
-            void _accept(const XCondition *condition) override
-            {
-                (*condition)[0]->visit(*this);
+            void _accept(const SimpleQuantifierCondition *condition) override {
+                condition->getCond()->visit(*this);
             }
 
             void _accept(const UntilCondition *condition) override
             {
-                (*condition)[0]->visit(*this);
-                (*condition)[1]->visit(*this);
+                condition->getCond1()->visit(*this);
```

Comment:
Keep the old `[]` notation instead, it would be nice to cut down on the number of specialized getters - in particular when they carry no significant information like `getCond1()`